probe-rs/src/flashing/loader.rs

```diff
@@ -343,7 +342,6 @@ impl FlashLoader {
                     options.keep_unwritten_bytes,
                     do_use_double_buffering,
                     options.skip_erase || do_chip_erase,
-                    options.progress.unwrap_or(&FlashProgress::new(|_| {})),
```

Comment:
I actually preferred this to the `Option` that is now used, then you don't have to check for the option on every call. But that's more a stylistic choice.